### PR TITLE
added treelite and simple example script

### PR DESCRIPTION
**Pull request by marsupialtail**
Added treelite benchmark to score.py

Will add Intel DAAL in a separate PR. 

Also added a simple example with small configurations that can be executed with a reasonable amount of RAM. Sklearn make_regression requires exorbitant amount of RAM.

### CODE REVIEW

File: src/scripts/lightgbm_python/score.py [hunk 1]
```diff
@@ -95,6 +97,21 @@ def run(args, other_args=[]):
     with metrics_logger.log_time_block("inferencing"):
         booster.predict(data=inference_raw_data, predict_disable_shape_check=bool(args.predict_disable_shape_check))
 
+    import pandas as pd
```
Comment (jfomhover) [inline src/scripts/lightgbm_python/score.py:100]: Let's split each framework into a distinct script. This way, we can maintian a collection of independent scripts, each reporting their metrics.

Can you create your score.py under the directory src/scripts/treelite_python/ instead?

**Comment by jfomhover** [comment]
Creating distinct scripts for different framework will be easier to maintain.

I'd suggest we create a script for each of the main tree learners, so we can run them against one another.

File: simple_example.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (jfomhover) [inline simple_example.sh:1]: If that's ok, I'd rather keep this kind of sequence in the instructions first. I'm working on creating pipelines based on each of the scripts, we could use a more consistent and configurable approach for running the whole thing locally.

**Review by jfomhover — CHANGES_REQUESTED** [review]
According to the comments left in the PR, could you:
- create a distinct script for treelite under /src/scripts/treelite ?
- remove the bash script (for now)

**Comment by marsupialtail** [comment]
I will separate the treelite code into a separate script -- the danger would be that the data loading code is the same, and if someone wants to change that in the future will have some code replication.

File: src/scripts/lightgbm_python/score.py [hunk 3]
```diff
@@ -42,6 +42,8 @@ def get_arg_parser(parser=None):
     group_i = parser.add_argument_group("Input Data")
     group_i.add_argument("--data",
         required=True, type=input_file_path, help="Inferencing data location (file path)")
+    group_i.add_argument("--nthreads",
```
Comment (jfomhover) [inline src/scripts/lightgbm_python/score.py:45]: can you roll this change back? I believe this was intended for the treelite score.py

File: src/scripts/treelite_python/score.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    metrics_logger = MetricsLogger("treelite.score")
+
+    # add some properties to the session
+    metrics_logger.set_properties(
+        framework = 'treelite_python',
+        task = 'score',
+        lightgbm_version = lightgbm.__version__
+    )
+
+    import pandas as pd
```
Comment (jfomhover) [inline src/scripts/treelite_python/score.py:81]: Could you add those in the header section of the code?

This way, we can catch missing dependencies during unit tests

**Review by jfomhover — CHANGES_REQUESTED** [review]

**Review by jfomhover — APPROVED** [review]
Thanks for the changes!

**Pull request merged**